tensorflow/compiler/xla/service/algebraic_simplifier.cc

```diff
@@ -80,6 +80,51 @@ bool IsAll(const HloInstruction* op, int8 value) {
   }
 }
 
+bool IsPositive(const HloInstruction* hlo,
+                const AlgebraicSimplifierOptions& options) {
+  switch (hlo->opcode()) {
+    case HloOpcode::kGetTupleElement: {
+      const HloInstruction* gte_operand = hlo->operand(0);
+      switch (gte_operand->opcode()) {
+        case HloOpcode::kCustomCall: {
+          const auto& target = gte_operand->custom_call_target();
+          return target ==
+                     options.get_cudnn_batchnorm_forward_training_metadata() &&
+                 hlo->tuple_index() == 2;
+        }
+        default:
+          return false;
+      }
+    }
+    case HloOpcode::kPower:
```

Comment:
For kPower, we need to make sure that the second operand is not a complex number, right?

Reply:
yea...looks like TF 2.0 allows complex for both operands. Not sure abt the semantics of TF 1.0 though. Anyway, I'll mk the change.

Reply:
Even operand(0) should not be complex. Only a pure imaginary number can generate a real number (that too negative). So it doesn't make sense to allow any operand to be complex.